### ISSUE TITLE
Ability to edit itself

### ISSUE THREAD

**Issue by 1advent** [CryptoDevWill/ArcAngelGPT#12]
This is dangerous, so you may want to not do it, but I want to add the ability of chatgpt to edit the code that makes up arc, thereby letting it add its own abilities.

**Comment by zenchantlive** [comment]
> This is dangerous, so you may want to not do it, but I want to add the ability of chatgpt to edit the code that makes up arc, thereby letting it add its own abilities.

If done with only human aporoval, maybe?

**Comment by CryptoDevWill** [comment]
> 

Yeah, we can have those checks in place

**Comment by zenchantlive** [comment]
if we do I supposed implementing guidelines within its code would be apropos

**Comment by CryptoDevWill** [comment]
> 

I agree, were you able to test out the new V2? it's on the main branch. We added the web scraper. Use the main branch.. I gotta fix the dev branch.
